presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsMetadata.java

```diff
@@ -64,13 +64,16 @@ public class TpcdsMetadata
     private final Set<String> tableNames;
     private final TpcdsTableStatisticsFactory tpcdsTableStatisticsFactory = new TpcdsTableStatisticsFactory();
 
-    public TpcdsMetadata()
+    private final boolean toggleCharToVarchar;
+
+    public TpcdsMetadata(boolean toggleCharToVarchar)
```

Comment:
We can retain the old constructor and have it set toggleCharToVarchar as false, so that it can be used by the Java connector. Then you will not need to change tests.